Rubberduck.CodeAnalysis/Inspections/Concrete/UnreachableCaseInspection/ParseTreeValueVisitor.cs

```diff
@@ -268,31 +294,58 @@ private void GetContextValue(ParserRuleContext context, out string declaredTypeN
                 expressionValue = rangeClauseIdentifierReference.IdentifierName;
                 declaredTypeName = GetBaseTypeForDeclaration(declaration);
 
-                if (declaration.DeclarationType.HasFlag(DeclarationType.Constant)
-                    || declaration.DeclarationType.HasFlag(DeclarationType.EnumerationMember))
+                if (declaration.DeclarationType.HasFlag(DeclarationType.Constant))
+                {
+                    expressionValue = GetConstantContextValueToken(declaration.Context);
+                    if (declaration.DeclarationType.HasFlag(DeclarationType.Constant)
+                        && declaredTypeName.Equals(Tokens.String))
+                    {
+                        expressionValue = "\"" + expressionValue + "\"";
+                    }
+                }
+                else if (declaration.DeclarationType.HasFlag(DeclarationType.EnumerationMember))
                 {
-                    expressionValue = GetConstantDeclarationValueToken(declaration);
+                    declaredTypeName = Tokens.Long;
+                    expressionValue = GetConstantContextValueToken(declaration.Context);
+                    if (expressionValue.Equals(string.Empty))
+                    {
+                        var enumValues = _enumMembers.Where(dt => dt.ConstantContext == declaration.Context);
+                        if (enumValues.Any())
+                        {
+                            var enumValue = enumValues.First();
+                            expressionValue = enumValue.Value.ToString();
+                        }
+                    }
                 }
             }
         }
 
         private bool TryGetIdentifierReferenceForContext(ParserRuleContext context, out IdentifierReference idRef)
         {
             idRef = null;
-            var identifierReferences = (_state.DeclarationFinder.MatchName(context.GetText()).Select(dec => dec.References)).SelectMany(rf => rf)
-                .Where(rf => rf.Context == context);
-            if (identifierReferences.Count() == 1)
+            if (IdRefRetriever != null)
             {
-                idRef = identifierReferences.First();
-                return true;
+                (bool success, IdentifierReference idReference) = IdRefRetriever(context);
+                idRef = idReference;
+                return success;
+            }
+            else if (State != null) //State is set to non-null for testing
```

Comment:
I don't understand why the state is needed only for the testing. Can we avoid having different code path between the production and testing?

Reply:
Agreed - removed.